### PR TITLE
feat: add pipeline_crons soft_delete_at

### PR DESCRIPTION
**Pull request by Malyue**
#### What this PR does / why we need it:
To add a soft delete field to the pipeline_crons table for easier recovery in case of accidental deletion.

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=569961&iterationID=12783&type=TASK)

#### Specified Reviewers:

/assign @sfwn

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  feat: add pipeline-crons soft_delete_at            |
| 🇨🇳 中文    |  新增定时流水线表的软删除字段            |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: .erda/migrations/pipeline/20240221-pipeline-crons-add-softdelete.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE `pipeline_crons` ADD COLUMN `soft_deleted_at` bigint(20) NOT NULL DEFAULT '0' COMMENT '软删除';
```
Comment (sfwn) [inline .erda/migrations/pipeline/20240221-pipeline-crons-add-softdelete.sql:1]: add `after xxx` to ensure new field location.

File: internal/tools/pipeline/providers/cron/db/define.go [hunk 2]
```diff
@@ -57,7 +57,8 @@ type PipelineCron struct {
 	// cluster name
 	ClusterName string `json:"clusterName" xorm:"cluster_name"`
 
-	IsEdge *bool `json:"is_edge"`
+	IsEdge        *bool  `json:"is_edge"`
+	SoftDeletedAt uint64 `json:"softDeletedAt" xorm:"deleted"`
```
Comment (sfwn) [inline internal/tools/pipeline/providers/cron/db/define.go:61]: Put `SoftDeletedAt` after `TimeUpdated`.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: .erda/migrations/pipeline/20240221-pipeline-crons-add-softdelete.sql (hunk 1 above)
Comment (Malyue) [reply .erda/migrations/pipeline/20240221-pipeline-crons-add-softdelete.sql:1]: Done.

File: internal/tools/pipeline/providers/cron/db/define.go (hunk 2 above)
Comment (Malyue) [reply internal/tools/pipeline/providers/cron/db/define.go]: Done.

**Pull request merged**